libraries/botbuilder-core/src/testAdapter.ts

```diff
@@ -43,120 +43,255 @@ export type TestActivityInspector = (activity: Partial<Activity>, description: s
  * ```
  */
 export class TestAdapter extends BotAdapter implements ExtendedUserTokenProvider {
+    /**
+     * Creates a new TestAdapter instance.
+     * @param logicOrConversation The bots logic that's under test.
+     * @param template (Optional) activity containing default values to assign to all test messages received.
+     */
+    public constructor(logicOrConversation?: ((context: TurnContext) => Promise<void>) | ConversationReference, template?: Partial<Activity>, sendTraceActivity = false) {
+        super();
+        this._sendTraceActivity = sendTraceActivity;
+        this.template = template || {};
+        if (logicOrConversation) {
+            if (typeof (logicOrConversation) === 'function') {
+                this._logic = logicOrConversation;
+                this.conversation = TestAdapter.createConversation('Convo1');
+            } else {
+                this.conversation = logicOrConversation;
+            }
+        } else {
+            this.conversation = TestAdapter.createConversation('Convo1');
+        }
+
+        this.conversation.locale = this.conversation.locale || this.locale;
+        if (this.template.locale) { this.conversation.locale = this.template.locale; }
+        if (this.template.serviceUrl) { this.conversation.serviceUrl = this.template.serviceUrl; }
+        if (this.template.channelId) { this.conversation.channelId = this.template.channelId; }
+        if (this.template.recipient) { this.conversation.bot = this.template.recipient; }
+        if (this.template.from) { this.conversation.user = this.template.from; }
```

Comment:
Consider `Object.assign`, though if falsy values are problematic you may need to take special care.